### PR TITLE
Redirect from Home Page to Greenbelt User Posts

### LINKED ISSUE #29: Write redirect function to hit the 'Posts' page when the user checks in to a location at the home page.
We added a 'check in' component to the home page, I will need to research whether this will leverage react router or function as a standard button href.

### PR DESCRIPTION
**Pull request by jctobey**
This code contains a redirect to the user posts page when the user clicks on 'check in at location'. In addition, it's got a loading gif that will display if the 'checkedIn' state=true.

Also added logic for 'login' or 'logout' buttons. Somehow, the CSS got lost that was spreading out the nav bar contents. Not sure what happened here, but please review in the PR.

Fixes #29

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,66 +1,20 @@
...
 import AppBar from "@material-ui/core/AppBar";
 import Toolbar from "@material-ui/core/Toolbar";
 import Typography from "@material-ui/core/Typography";
 import Button from "@material-ui/core/Button";
 import IconButton from "@material-ui/core/IconButton";
 import MenuIcon from "@material-ui/icons/Menu";
-import Menu from '@material-ui/core/Menu';
-import MenuItem from '@material-ui/core/MenuItem';
-import ListItemIcon from '@material-ui/core/ListItemIcon';
-import ListItemText from '@material-ui/core/ListItemText';
```
Comment (themarcusaurelius) [inline client/src/App.js]: This is for the menu items in the hamburger drop down. We will need this if we are keeping the hamburger.

File: client/src/App.js [hunk 2]
```diff
@@ -1,66 +1,20 @@
...
-
-//Material UI Icons for Menu
-import AccountBalance from "@material-ui/icons/AccountBalance";
-import LocationOn from "@material-ui/icons/LocationOn";
-import Fingerprint from "@material-ui/icons/Fingerprint";
-import Person from "@material-ui/icons/Person";
```
Comment (themarcusaurelius) [inline client/src/App.js]: These are the icons for the Hamburger bar drop down as well.

File: client/src/App.js [hunk 3]
```diff
@@ -138,78 +91,38 @@ class App extends Component {
           </Navbar.Header>
         </Navbar> */}
         {withStyles}
-
-        <div className={classes.root}>
```
Comment (themarcusaurelius) [inline client/src/App.js]: These are dependent on the styles above. should be classes.root. I remember fixing this already last time the styling got messed up.

File: client/src/App.js [hunk 4]
```diff
@@ -138,78 +91,38 @@ class App extends Component {
...
+        <div className={styles.root}>
```
Comment (themarcusaurelius) [inline client/src/App.js]: classes.root this will fix the nav layout

File: client/src/App.js [hunk 5]
```diff
@@ -138,78 +91,38 @@ class App extends Component {
...
           <AppBar position="static">
             <Toolbar className="theme">
-              <Typography variant="h6" color="inherit" className={classes.grow}>
+              <Typography variant="h6" color="inherit" className={styles.grow}>
```
Comment (themarcusaurelius) [inline client/src/App.js]: same as before.   `classes.grow`

File: client/src/App.js [hunk 6]
```diff
@@ -138,78 +91,38 @@ class App extends Component {
...
                 className={styles.menuButton}
                 color="inherit"
                 aria-label="Menu"
-                aria-owns={anchorEl ? 'simple-menu' : undefined}
-                aria-haspopup="true"
-                onClick={this.handleClick}
               >
                 <MenuIcon />
               </IconButton>
-              <Menu
```
Comment (themarcusaurelius) [inline client/src/App.js]: This is everything contained in the Hamburger drop down menu

File: client/src/App.js [hunk 7]
```diff
@@ -138,78 +91,38 @@ class App extends Component {
...
-    )
+    );
   }
 }
 
-App.propTypes = {
-  classes: PropTypes.object.isRequired,
-};
-
-export default withStyles(styles)(App);
```
Comment (themarcusaurelius) [inline client/src/App.js]: Need withStyles here for all the styling to be exported.

File: client/src/App.js [hunk 8]
```diff
@@ -1,66 +1,20 @@
...
-import MenuItem from '@material-ui/core/MenuItem';
-import ListItemIcon from '@material-ui/core/ListItemIcon';
-import ListItemText from '@material-ui/core/ListItemText';
-
-//Material UI Icons for Menu
-import AccountBalance from "@material-ui/icons/AccountBalance";
-import LocationOn from "@material-ui/icons/LocationOn";
-import Fingerprint from "@material-ui/icons/Fingerprint";
-import Person from "@material-ui/icons/Person";
-
```
Comment (themarcusaurelius) [inline client/src/App.js]: This is the styling for the nav bar that was deleted last time the layout got messed up. We will need to keep this as well

File: client/src/App.js [hunk 9]
```diff
@@ -1,66 +1,20 @@
...
-
-const styles = theme => ({
```
Comment (themarcusaurelius) [inline client/src/App.js]: This is the styling for the nav bar that was deleted last time the layout got messed up. We will need to keep this as well

File: client/src/components/Container/Container.jsx [hunk 10]
```diff
@@ -62,34 +60,36 @@ function PaperSheet(props) {
...
-                        className={classes.button}
-                        onClick={() => props.handleCheckIn()}
-                        id={id}
-                      >
-                        <CheckIn className={classes.leftIcon} />
-                        Check in at: {name}
-                      </Button>
-                    </div>
-                  ) : (
+
```
Comment (themarcusaurelius) [inline client/src/components/Container/Container.jsx]: This looks pretty cool. Excited to see the GIf

File: client/src/index.js [hunk 11]
```diff
@@ -1,8 +1,8 @@
 import ReactDOM from "react-dom";
+// import "./index.css";
```
Comment (themarcusaurelius) [inline client/src/index.js]: Not sure we need these anymore

File: client/src/pages/Main.jsx [hunk 12]
```diff
@@ -132,8 +132,11 @@ class Main extends Component {
   // Constructor and state
   state = {
     currentAccessPoint: {},
-    checkInLocation: {}
```
Comment (themarcusaurelius) [inline client/src/pages/Main.jsx]: Cool

File: client/src/pages/Main.jsx [hunk 13]
```diff
@@ -160,12 +163,19 @@ class Main extends Component {
...
+      checkedIn: true
     });
+    this.handleRedirect = setTimeout(() => {
+      this.setState(() => ({ toPosts: true }));
+    }, 2000);
   };
 
   // Render function
   render() {
+    if (this.state.toPosts === true) {
```
Comment (themarcusaurelius) [inline client/src/pages/Main.jsx]: Cool

File: client/src/routes.js [hunk 14]
```diff
@@ -53,6 +53,11 @@ export const makeMainRoutes = () => {
             path="/profile"
             render={props => <Profile auth={auth} {...props} />}
           />
+          <Route
```
Comment (themarcusaurelius) [inline client/src/routes.js]: Beautiful

File: client/src/routes.js [hunk 15]
```diff
@@ -16,14 +17,13 @@ const theme = createMuiTheme({
       contrastText: "#fff"
     },
     secondary: {
-      light: "#6fbf73",
+      light: "#ff7961",
```
Comment (themarcusaurelius) [inline client/src/routes.js]: This got switched too

**Comment by jctobey** [comment]
Gonna try getting this branch up to date and re-submitting

**Pull request closed without merging**